packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -176,6 +185,7 @@ export class AppSyncModelVisitor<
       scalars: buildScalars(_schema, rawConfig.scalars || '', defaultScalars),
       target: rawConfig.target,
       isTimestampFieldsAdded: rawConfig.isTimestampFieldsAdded,
+      generateIndexRules: rawConfig.generateIndexRules,
```

Comment:
If this config is only applied to swift, you can parse this config in the constructor of `appsync-swift-visitor` rather than the base one. A example from the js visitor: https://github.com/aws-amplify/amplify-codegen/blob/bdde4bd646209d08de96d6079c58d65bddf64d45/packages/appsync-modelgen-plugin/src/visitors/appsync-javascript-visitor.ts#L37-L45